Environment variables can't be used in config files

Jutanium opened the pull request:
<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->

#### What kind of changes does this PR include?
<!-- Delete any that don’t apply -->

- New or updated content

#### Description

We closed #785 as the corresponding feature was closed, but the note about Astro being unable to load env variables is still important information, even if we don't have a workaround. This PR restores that note and adds a corresponding one to the environmental variables change.

I could see us only wanting one of those two messages, or rather wanting this to be in Troubleshooting. Let me know!

@natemoo-re, if there's currently a workaround to access env variables in Astro, let me know.

<!--
Here’s what will happen next:

1. Our GitHub bots will run to check your changes.
   If they spot any broken links you will see some error messages on this PR.
   Don’t hesitate to ask any questions if you’re not sure what these mean!

2. In a few minutes, you’ll be able to see a preview of your changes on Netlify 🥳

3. One or more of our maintainers will take a look and may ask you to make changes.
   We try to be responsive, but don’t worry if this takes a day or two.
-->

delucis reviewed (commented):
You can access environment variables using `process.env` in the config file, with the caveat that `.env` files are NOT loaded yet. An environment variable set by a CI service or on the CLI _will_ be accessible though. For example, you could use `process.env.NETLIFY` to toggle a feature only in Netlify builds or use `TOKEN=secrets npm run dev` to expose `process.env.TOKEN`.

I think I've also heard people are using `dotenv` to load their `.env` files, so that could be a workaround if it doesn't have drawbacks — Nate would know best!

delucis commented:
No, `import.meta.env` is a Vite thing and the config is evaluated in a standard Node environment before Vite is running.

Jutanium commented:
Added a section on setting variables with the CLI.

I moved the caution in Environment Variables to the overview. It really is an overview, not an introduction, and `import.meta.env` is already brought up by that point.

I wasn't sure whether to document `process.env` in Environment Variables or Configuring Astro. I chose the later because Configuring Astro seems to want to be comprehensive, but I could see other possibilities:
- I could do the opposite where I just have a section in Environment Variables that has all the info, and don't change Configuring Astro. 
- Or I could have a caution in Configuring Astro that links to Environment Variables, where the info actually lies.

delucis commented on src/pages/en/guides/environment-variables.md:
```diff
@@ -16,6 +17,10 @@ PUBLIC_ANYBODY=there
+`import.meta.env` and `.env` files are not available inside [configuration files](/en/guides/configuring-astro/#environment-variables). 
+:::
+
 ## Default environment Variables
```
Consistent casing:

```suggestion
## Default environment variables
```

delucis reviewed (commented):
I think these are good additions, Dan — thank you!

It’s tricky deciding where to document `process.env` but I think I agree with where you ended up — the most common use case is in the astro config file so makes sense to document it there.

I could leave a bunch more comments on the _rest_ of the environment variables page but I’ll hold off until you say that would be helpful.

delucis commented:
More comments as requested by @Jutanium (to be added to this PR or tackled later at his discretion).

1. Opening paragraph:

   > Astro uses Vite for environment variables

   I’d probably rewrite to something like
   
   > Astro relies on Vite’s built-in environment variable support
   
   Or even rewrite the whole opening sentence:
   
   > Astro lets you [use any of Vite’s methods](https://vitejs.dev/guide/env-and-mode.html) to get and set environment variables.

2. “Default environment variables” section
   
   Totally a nit but in the first list item I’d write either “`development | production`” or “`development` or `production`” but not use the `|` outside of `<code>`.

2. “Supported file names” code snippet

   Could probably considered a nit, but might reformat as a table or list. At first glance I read it as a continuation of the `.env` file example above it.

4. “Getting environment variables” section

   The browser support tip seems excessively noisy in this context. I’d roll that content into the opening paragraph.

Jutanium commented:
Thanks @delucis! This page could use a broader reworking (including explaining what Environment Variables actually _are_), but I made minimal changes for now to address your points.

I removed the .env files snippet entirely as it was copy-pasted from Vite docs - I linked to that instead. I also removed the the tip rather than moving it, as we made this point clear as soon as we talked about client-side code in the intro.

delucis reviewed (approved):
Thanks Dan! LGTM 🙌
